addons/l10n_id/models/res_bank.py

```diff
@@ -87,7 +87,7 @@ def _get_qr_vals(self, qr_method, amount, currency, debtor_partner, free_communi
                 "do": "create-invoice",
                 "apikey": self.l10n_id_qris_api_key,
                 "mID": self.l10n_id_qris_mid,
-                "cliTrxNumber": structured_communication,
+                "cliTrxNumber": free_communication if self._context.get('is_online_qr') else structured_communication,
```

Comment:
```suggestion
                "cliTrxNumber": structured_communication or free_communication,
```
`is_online_qr` is always True if we get to this point (of generating QRIS QR), so I think we can omit it.
Also by doing it this way, we can retain `structured_communication` format for invoice while fixing the one for PoS.
Thanks for the suggestion @vin-odoo